components/brave_news/browser/brave_news_controller.cc

```diff
@@ -455,39 +458,85 @@ void BraveNewsController::GetImageData(const GURL& padded_image_url,
   const std::string ending = ".pad";
   if (file_name.length() >= file_name.max_size() - 1 ||
       file_name.length() <= ending.length()) {
-    std::optional<std::vector<uint8_t>> args;
-    std::move(callback).Run(std::move(args));
+    std::move(callback).Run(std::nullopt);
     return;
   }
   const bool is_padded =
       (file_name.compare(file_name.length() - ending.length(), ending.length(),
                          ending) == 0);
   VLOG(3) << "is padded: " << is_padded;
+  // Ensure we don't de-pad non PCDN images
+  if (is_padded && !padded_image_url.DomainIs(GetMatchingPCDNHostname())) {
+    DLOG(ERROR) << "Encountered a padded URL apparently not from PCDN: "
+                << padded_image_url.spec();
+    std::move(callback).Run(std::nullopt);
+    return;
+  }
   // Make the request
   private_cdn_request_helper_.DownloadToString(
       padded_image_url,
+      base::BindOnce(&BraveNewsController::OnGetImageDataFetched,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback),
+                     is_padded));
+}
+
+void BraveNewsController::OnGetImageDataFetched(GetImageDataCallback callback,
+                                                const bool is_padded,
+                                                const int response_code,
+                                                const std::string& body) {
+  // Handle the response
+  VLOG(3) << "getimagedata response code: " << response_code;
+  // Attempt to remove byte padding if applicable
+  std::string_view body_payload(body.data(), body.size());
+  if (response_code < 200 || response_code >= 300 ||
+      (is_padded &&
+       !brave::PrivateCdnHelper::GetInstance()->RemovePadding(&body_payload))) {
+    // Byte padding removal failed
+    std::move(callback).Run(std::nullopt);
+    return;
+  }
+  // Download (and optional unpadding) was successful
+  std::vector<uint8_t> image_bytes(body_payload.begin(), body_payload.end());
+#if defined(OS_ANDROID)
+  std::move(callback).Run(std::move(image_bytes));
+#else
+  if (is_padded) {
```

Comment:
We don't reencode when the image is padded, because we already reencode the images on our CDN, right?

Maybe worth adding a comment saying why we don't need to decode/reencode here?